bluetooth_mesh/messages/silvair/test/test_rrule_scheduler.py

```diff
@@ -0,0 +1,187 @@
+                      {"rule_id": RuleIDs.BYMINUTE, "rule": [0x01, 0x0B]},
+                      {"rule_id": RuleIDs.BYHOUR, "rule": [0x02, 0x0C, 0x10]},
+                      {"rule_id": RuleIDs.BYDAY, "rule":
+                          [Days.NEXT_MONDAY, Days.NEXT_FRIDAY, Days.NEXT_SUNDAY, Days.PREVIOUS_SATURDAY]},
+                      {"rule_id": RuleIDs.BYMONTHDAY, "rule": [0x04, 0x10, 0x11, 0x12, 0x13]},
+                      {"rule_id": RuleIDs.BYYEARDAY, "rule":
+                          [0x0004, 0x0010, 0x0002, 0x0030, 0x0040, 0x0050, 0x0100, 0x0120]},
+                      {"rule_id": RuleIDs.BYWEEKNO, "rule": [0x06, 0x16, 0x26, 0x2A]},
+                      {"rule_id": RuleIDs.BYMONTH, "rule": 0x0201},
+                      {"rule_id": RuleIDs.BYSETPOS, "rule": [0x000A, 0x0010, 0x00A0, 0x00D0, 0x0050, 0x0150]},
+                      {"rule_id": RuleIDs.DTSTART, "rule":
+                          Container(second=56, minute=17, hour=22, day=25, month=11, year=2040)},
+                      {"rule_id": RuleIDs.UNTIL, "rule":
+                          Container(second=56, minute=17, hour=22, day=25, month=11, year=2040)},
+                      {"rule_id": RuleIDs.EXPLICIT_RDATE, "rule": [
+                          Container(second=56, minute=17, hour=22, day=25, month=11, year=2020),
+                          Container(second=56, minute=17, hour=22, day=25, month=11, year=2030),
+                          Container(second=56, minute=17, hour=22, day=25, month=11, year=2040)]},
+                      {"rule_id": RuleIDs.EXCLUSIONS_RDATE, "rule": [
+                          Container(second=56, minute=17, hour=22, day=25, month=11, year=2020),
+                          Container(second=56, minute=17, hour=22, day=25, month=11, year=2030),
+                          Container(second=56, minute=17, hour=22, day=25, month=11, year=2040)]}
+                  ]}
+                 )
+]
+# fmt: on
+
+
+@pytest.mark.parametrize(
+    "encoded, subopcode, payload", valid
+)
+def test_parse_schedule_register_entry_format(encoded, subopcode, payload):
+    decoded = RRuleSchedulerSubMessage.parse(encoded)
+    assert decoded.subopcode == subopcode
+    print(decoded)
+
+    assert decoded.payload.entry_id == (payload["entry_id"])
+
+    for idx, sl in enumerate(decoded.payload.scheduler_register_entry.array_of_slots):
+        assert sl["slot_id"] == payload["slots"][idx]["slot_id"]
+        assert sl["element"] == payload["slots"][idx]["element"]
+        assert sl["slot_parameter"] == payload["slots"][idx]["slot_parameter"]
+
+    for idx, rl in enumerate(decoded.payload.scheduler_register_entry.rules):
+        assert rl["rule_id"] == payload["rules"][idx]["rule_id"]
+        assert rl["rule"] == payload["rules"][idx]["rule"]
+
+
+def test_scheduler_rules_list_status():
+    encoded = b'\x01\x01\x02\x03\x04\x05'
+    payload = [RuleIDs.UNTIL, RuleIDs.COUNT, RuleIDs.INTERVAL, RuleIDs.BYSECOND, RuleIDs.BYMINUTE]
```

Comment:
I'd rather have this parse as a `set`, so that order and repetitions don't matter